### PR TITLE
LCAM-1611|Replace WebClient with Declarative HTTP Interface

### PR DESCRIPTION
**Pull request by venkat980**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-1611)

Refactored the service to use Declarative HTTP Interface instead of WebClient.
Using resillience4j for retry mechanism.
Updated spring framework and dependencies to latest version to address the snyk reported security vulnerabilities.
Fixed the date field types on the CreateContributionRequest.
Also, enhanced the tests to use stubbing and verify the responses.
Updated test classes to use the RequestBuilderUtils from crime-commons jar and deleted the local RequestBuilderUtils class to reduce code duplication.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

## Additional checks

- Don’t forget to [run](https://github.com/ministryofjustice/laa-crimeapps-maat-functional-tests/actions/workflows/ExecuteUiTests.yaml) the MAAT functional test suite after deploying your changes to the DEV or TEST environments to ensure your changes haven’t broken any of the functional tests.

### CODE REVIEW

File: crown-court-contribution/src/main/resources/application.yaml [hunk 1]
```diff
@@ -60,22 +60,37 @@ spring:
...
+      default:
+        max-attempts: 3
+        wait-duration: 2s
+        enableExponentialBackoff: true
+        exponentialBackoffMultiplier: 2
+        retry-exceptions:
+          - org.springframework.web.reactive.function.client.WebClientRequestException
+          - org.springframework.web.reactive.function.client.WebClientResponseException.BadGateway
+          - org.springframework.web.reactive.function.client.WebClientResponseException.TooManyRequests
+          - org.springframework.web.reactive.function.client.WebClientResponseException.InternalServerError
```
Comment (Mhowell494) [inline crown-court-contribution/src/main/resources/application.yaml:78]: Can we make sure this aligns with the changes in the Hardship service as I know we removed the `InternalServerError` in that service?

File: crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/controller/ContributionControllerContributionIntegrationTest.java [hunk 2]
```diff
@@ -205,45 +197,49 @@ void givenMaatApiException_whenCalculateContributionIsInvoked_thenInternalServer
...
+                .build();
+        wiremock.stubFor(get(urlEqualTo(summariesUrl))
                 .willReturn(
                         WireMock.aResponse()
                                 .withStatus(NOT_IMPLEMENTED.code())
+                                .withBody(Json.write(errorDTO))
                                 .withHeader("Content-Type", String.valueOf(MediaType.APPLICATION_JSON))
                 )
         );
         mvc.perform(buildRequestGivenContent(HttpMethod.GET, "", GET_CONTRIBUTION_SUMMARIES_ENDPOINT_URL))
```
Comment (Mhowell494) [inline crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/controller/ContributionControllerContributionIntegrationTest.java:232]: It looks like the `buildRequestGivenContent` method is available in `crime-commons-classes` module, can we use the version please? That way we can clean up the `RequestBuilderUtils` class in this service.

**Review by Mhowell494 — CHANGES_REQUESTED** [review]

File: crown-court-contribution/src/main/resources/application.yaml (hunk 1 above)
Comment (venkat980) [reply crown-court-contribution/src/main/resources/application.yaml:78]: Updated to be in line with Hardship.

File: crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/controller/ContributionControllerContributionIntegrationTest.java (hunk 2 above)
Comment (venkat980) [reply crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/controller/ContributionControllerContributionIntegrationTest.java:232]: Done. Updated the PR description to reflect the change.

**Review by Mhowell494 — APPROVED** [review]

**Pull request merged**